### PR TITLE
Feature/default profile picture

### PR DESCRIPTION
**Pull request by GhaliBerbich**
This pull request introduces several enhancements to profile image handling in our app, improving user experience by allowing more interactive and responsive interactions with profile pictures. Key updates include implementing default avatars, enabling immediate updates in the editing screen, and providing functionality to remove a profile picture.

Main Changes:
- Default Profile Picture: Adjusted the ProfileImage function to display a default picture when no specific image is found in storage. This ensures that all user profiles have a visual representation, even without a custom uploaded image.

- Deleting Profile Pictures: Introduced a new function deleteProfilePicture that allows users to remove their profile picture via the profile editing interface, enhancing user control over their profile content.
- Immediate Updates in Editing: Implemented logic that updates the profile picture immediately in the editing screen upon changes, allowing users to see how their new profile picture will look before saving changes.
- Default Avatars: Added a set of default avatars to the drawable folder, providing a variety of fallback images that enhance the aesthetic consistency across the app.
- Bug Fix: Removed a redundant test tag in the ProfileImage Box that was causing failures in automated tests.

Additional Notes:
- Refresh Issue: The profile screen still requires a manual refresh to display the updated image after changes are saved. This issue will be addressed in a subsequent update.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/camera/Gallery.kt [hunk 1]
```diff
@@ -53,24 +73,86 @@ fun ProfileImage(userId: String, modifier: Modifier = Modifier, imageViewModel:
...
   }
 
+  LaunchedEffect(bitmap) {
+    if (bitmap != null && editing) {
+      isImageRemoved = false
+    } else if (bitmap == null && imageUrl == null && editing) {
+      isImageRemoved = true
+    }
+  }
+  LaunchedEffect(imageUrl) {
```
Comment (elvan03) [inline app/src/main/java/com/android/sample/ui/camera/Gallery.kt:83]: the 2 launchedEffect blocks can be combined into 1 here :)
for example: 
isImageRemoved = when {
    bitmap != null && editing -> false
    imageUrl == null && bitmap == null && editing -> true
    else -> isImageRemoved
  }
  In this case, don't forget to give LaunchedEffect the 2 parameters, bitmap & imageUrl

**Review by elvan03 — CHANGES_REQUESTED** [review]
I have requested 1 improvement to reduce redundancy. But otherwise, everything looks clean to me! Thanks for commenting some parts of your code and the detailed explanation of your PR, it is helpful!

File: app/src/androidTest/java/com/android/sample/ui/profile/ProfileScreenTests.kt [hunk 2]
```diff
@@ -107,11 +107,7 @@ class ProfileScreenTest {
     composeTestRule.onNodeWithTag("Cycling").assertIsDisplayed()
     composeTestRule.onNodeWithTag("Reading").assertIsDisplayed()
 
-    composeTestRule.onNodeWithTag("goBackButton").assertIsDisplayed()
     composeTestRule.onNodeWithTag("bottomNavigationMenu").assertIsDisplayed()
```
Comment (DAFERMOH) [inline app/src/androidTest/java/com/android/sample/ui/profile/ProfileScreenTests.kt:110]: Removed assertions and actions related to the goBackButton.

File: app/src/main/java/com/android/sample/model/image/ImageRepository.kt [hunk 3]
```diff
@@ -34,4 +34,6 @@ interface ImageRepository {
       onSuccess: (List<Bitmap>) -> Unit,
       onFailure: (Exception) -> Unit
   )
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/model/image/ImageRepository.kt:36]: Added a new method deleteProfilePicture to delete a profile picture from Firebase storage.

File: app/src/main/java/com/android/sample/model/image/ImageRepositoryFirestore.kt [hunk 4]
```diff
@@ -140,4 +140,17 @@ constructor(private val firestore: FirebaseFirestore, private val storage: Fireb
         },
         onFailure)
   }
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/model/image/ImageRepositoryFirestore.kt:142]: Added a new method deleteProfilePicture to delete a profile picture from Firebase storage.

File: app/src/main/java/com/android/sample/model/image/ImageViewModel.kt [hunk 5]
```diff
@@ -50,4 +50,8 @@ class ImageViewModel @Inject constructor(private val repository: ImageRepository
   ) {
     repository.fetchActivityImagesAsBitmaps(activityId, onSuccess, onFailure)
   }
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/model/image/ImageViewModel.kt:52]: Added a new method deleteProfilePicture to the ImageViewModel to delegate the deletion to the repository.

File: app/src/main/java/com/android/sample/ui/camera/Gallery.kt [hunk 6]
```diff
@@ -53,24 +73,86 @@ fun ProfileImage(userId: String, modifier: Modifier = Modifier, imageViewModel:
...
+                  .background(color = Color.Red, shape = CircleShape)) {
+            Icon(
+                imageVector = Icons.Default.Close,
+                contentDescription = "Remove Image",
+                tint = Color.White)
+          }
+    }
+  }
+}
+
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/camera/Gallery.kt:139]: Introduced a new function randomDefaultProfileImage to randomly select a default profile image.

File: app/src/main/java/com/android/sample/ui/profile/EditProfile.kt [hunk 7]
```diff
@@ -89,7 +89,7 @@ fun EditProfileScreen(
   val context = LocalContext.current
   val networkManager = NetworkManager(context)
   var selectedImage by remember { mutableStateOf<Bitmap?>(null) }
-
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/profile/EditProfile.kt:92]: Added state management for picture removal.

File: app/src/main/java/com/android/sample/ui/profile/EditProfile.kt [hunk 8]
```diff
@@ -122,17 +122,7 @@ fun EditProfileScreen(
         if (isGalleryOpen) {
           GalleryScreen(
               isGalleryOpen = { isGalleryOpen = false },
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/profile/EditProfile.kt:124]: Updated GalleryScreen and ProfileImage composables to handle image editing and removal.

File: app/src/main/java/com/android/sample/ui/profile/EditProfile.kt [hunk 9]
```diff
@@ -210,6 +196,18 @@ fun EditProfileScreen(
 
                 Button(
                     onClick = {
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/profile/EditProfile.kt:198]: Added logic to delete the profile picture from the ImageViewModel when saving changes.

File: app/src/main/java/com/android/sample/ui/profile/Profile.kt [hunk 10]
```diff
@@ -127,21 +126,11 @@ fun ProfileContent(
       topBar = {
         TopAppBar(
             title = { Text("Profile") },
```
Comment (DAFERMOH) [inline app/src/main/java/com/android/sample/ui/profile/Profile.kt:128]: Removed the goBackButton from the top app bar.

**Review by DAFERMOH — APPROVED** [review]
**Main Changes**
**Default Profile Picture:** Adjusted the ProfileImage function to display a default picture when no specific image is found in storage. This ensures that all user profiles have a visual representation, even without a custom uploaded image.
Deleting Profile Pictures: Introduced a new function deleteProfilePicture that allows users to remove their profile picture via the profile editing interface, enhancing user control over their profile content.
**Immediate Updates in Editing:** Implemented logic that updates the profile picture immediately in the editing screen upon changes, allowing users to see how their new profile picture will look before saving changes.
Default Avatars: Added a set of default avatars to the drawable folder, providing a variety of fallback images that enhance the aesthetic consistency across the app.
**Bug Fix:** Removed a redundant test tag in the ProfileImage Box that was causing failures in automated tests.
**Conclusion:** good job implementing default avatars, enabling immediate updates in the editing screen, and providing functionality to remove a profile picture. You can fix the issue where the profile screen still requires a manual refresh to display the updated image after changes are saved for the next sprint.

**Review by DAFERMOH — CHANGES_REQUESTED** [review]
you have some merge conflicts to solve

File: app/src/main/java/com/android/sample/ui/camera/Gallery.kt (hunk 1 above)
Comment (GhaliBerbich) [reply app/src/main/java/com/android/sample/ui/camera/Gallery.kt:83]: Thank you for your suggestion, however here using launched effect is about when to "activate" the clause. One will be activated when the imageUrl will be changed(if the picture is changed, or removed from the storage), and the other will be activated if a picture is put as a replacement of the current profile picture, before saving it. So I think that it is important to keep these 2 blocks as they treat independant cases.

Comment (elvan03) [reply app/src/main/java/com/android/sample/ui/camera/Gallery.kt:83]: Understood! I see that it's more logical if they're independent cases!

**Review by elvan03 — APPROVED** [review]
Thanks for your changes! LGTM!

**Review by DAFERMOH — APPROVED** [review]
Good job merging your code with latest changes in main.

**Pull request merged**